commcare_connect/opportunity/models.py

```diff
@@ -370,11 +425,41 @@ def images(self):
 
 class OpportunityClaim(models.Model):
     opportunity_access = models.OneToOneField(OpportunityAccess, on_delete=models.CASCADE)
-    max_payments = models.IntegerField()
+    # to be removed
+    max_payments = models.IntegerField(null=True)
     end_date = models.DateField()
     date_claimed = models.DateField(auto_now_add=True)
 
 
+class OpportunityClaimLimit(models.Model):
+    opportunity_claim = models.ForeignKey(OpportunityClaim, on_delete=models.CASCADE)
+    payment_unit = models.ForeignKey(PaymentUnit, on_delete=models.CASCADE)
```

Comment:
Is strictly on `OpportunityClaimLimit` per user per payment unit? If so, can that be enforced at the db level (a unique constraint on the claim/unit pair.)

Reply:
Yeah. It is per paymentunit/claim. I guess we can, though we were not doing the same sort of db constraint for `OpportunityClaim` previously. We are (and were) using a `get_or_create` based on claim/paymentunit.

Reply:
`OpportunityClaim` has a `OneToOneField` foreign key to `OpportunityAccess` which provides that constraint (there can only be one per user per opportunity).

Reply:
https://github.com/dimagi/commcare-connect/pull/298/files#diff-0718ce467ae87450bbb0b1d53245b67353a89cc48810cfb040f38a31e1bfd77f